include/fiction/utils/truth_table_utils.hpp

```diff
@@ -409,6 +411,24 @@ namespace fiction
     return std::vector<kitty::dynamic_truth_table>{table1, table2};
 }
 
+/**
+ * This function calculates the output index for a given input index by evaluating the truth table.
+ *
+ * @param truth_table The truth table to evaluate.
+ * @param current_input_index The index representing the current input pattern.
+ * @return The output index derived from the truth table for the given input index.
+ */
+[[nodiscard]] uint64_t determine_output_index_of_output(const std::vector<kitty::dynamic_truth_table>& truth_table,
+                                                        const uint64_t current_input_index) noexcept
+{
+    std::bitset<64> bits{};
+    for (auto i = 0u; i < truth_table.size(); i++)
+    {
+        bits[i] = (kitty::get_bit(truth_table[i], current_input_index) != 0u);
+    }
+    return bits.to_ulong();
```

Comment:
This looks like there should be a function in the `kitty` library implementing this already. Please check the documentation.